### DISCUSSION TITLE
Conditional Expression Not Triggering Correctly in GitHub Actions

### DISCUSSION THREAD

**Discussion by zeto-bit** [community/community#140804] [Actions] [answered]
### Select Topic Area

Question

### Body

I'm trying to run a GitHub Actions workflow where I use a conditional expression to upload Black Duck scan results only if the previous step succeeds. However, the upload step is being skipped, and I can't figure out why. Here's a snippet of my workflow

- name: Upload Scan Results
  if: success()              # Conditional to check for success
  uses: actions/upload-artifact@v3
  with:
    name: blackduck-results
    path: ./blackduck-output

I expected the Upload Scan Results step to run after the scan completes successfully, but it seems to be skipped without any clear error message.

Questions:
Is there something wrong with how I'm using the if: success() condition?
Could this be a syntax issue, or am I missing something about how GitHub Actions handles conditional expressions?
What is the correct way to write the conditional for this situation?

**Comment by AdarshBP** [comment, marked as answer]
The issue lies in the syntax of the if condition in GitHub Actions. In your snippet:

- name: Upload Scan Results
  if: ${{ success() }}    # Corrected conditional
  uses: actions/upload-artifact@v3
  with:
    name: blackduck-results
    path: ./blackduck-output
